### PR TITLE
Updates for Notes

### PR DESCRIPTION
**Pull request by D0cR3d**
Fixes many things in #65 

- Disabled ability to run command in DM
- Restrict access to command
  - **Vanguard**, **Discord Chat Mods** can do any command
  - **Mod Assistant** can do any command except delete
- Renamed `set` to `add`
- Updated some of the error messages to be more informative.
- Update Notes to use the `ModActions` table.
- Removed 255 character note constraint limit.
- Date now shows with the history command.
  - Currently set to show `1. Note Text (@ModName) [Date and Timestamp]` [Example](https://i.imgur.com/8BIlWDy.png)
- Notes must not be empty.
- Added ability to delete all notes.
  - TODO: Add confirmation before delete.
- Removed unnecessary keyword `note` from help text.
- Removed `?` from help text for argument as those are required.
- Add alias for `note` command
- Add alias for `add` and `history` commands

### CODE REVIEW

File: src/commands/profile/Note.ts [hunk 1]
```diff
@@ -43,78 +55,120 @@ export default class Note extends Command {
...
-					.then(results => {
-						if (!results.length) {
-							message.channel.send(`There are no notes for ${user.username}.`);
-							return message.channel.stopTyping();
-						} else {
-							let notes: string = '';
-							results.forEach((value: any, index: number) => {
-								notes += `${index + 1}. ${value.note} (<@${value.author}>)\u000d`;
-							});
+				if (author.roles.has(modRoles[0].id) || author.roles.has(modRoles[1].id) || author.roles.has(modRoles[2].id)) {
```
Comment (TheRealBGoat) [inline src/commands/profile/Note.ts]: Can all of these role checks be abstracted out into a separate function to improve readability?

```typescript
private hasPermission(author: GuildMember, action: string) : boolean { 
    // Switch on action and return boolean for check against whether the author has the necessary roles
}
```
Forgive any syntax errors in the somewhat pseudo-code above

File: src/commands/profile/Note.ts [hunk 2]
```diff
@@ -10,18 +10,30 @@ export default class Note extends Command {
...
 			name: 'note',
-			desc: 'Set a note for a user',
+			desc: 'Add a note for a user',
 			usage: '<prefix>note <Argument>?',
 			info: 'Argument information below...\u000d\u000d' +
-			'note set <User>             : Sets a note for specified user\u000d' +
-			'note history <User>         : Displays note history for specified user\u000d' +
-			'note delete <User> <NoteID> : Deletes specific note from specified user',
-			group: 'profile'
+			'note add <User> <Note>  : Adds a note for specified user\u000d' +
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: can remove `note` from this line, and the following 3.  This section describes arguments only.

File: src/commands/profile/Note.ts (hunk 1 above)
Comment (katagatame) [inline src/commands/profile/Note.ts]: I'm all for readability, Any changes to the code to improve that is welcome.

File: src/commands/profile/Note.ts [hunk 3]
```diff
@@ -10,18 +10,30 @@ export default class Note extends Command {
 	public constructor() {
 		super({
 			name: 'note',
-			desc: 'Set a note for a user',
+			desc: 'Add a note for a user',
 			usage: '<prefix>note <Argument>?',
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: note must have an argument, remove `?`.

**Comment by katagatame** [comment]
Here's a thought.

We'll be using this command a lot, and I imagine in the early usage we'd be typing everything out but down the road we'll get tired of typing out everything.

Maybe we some aliases?

so this
`.note add @katatagatame#1234 test`

becomes this
`.n a @katagatame#1234 test`

Thoughts?

File: src/commands/profile/Note.ts (hunk 2 above)
Comment (D0cR3d) [inline src/commands/profile/Note.ts]: I can do that.

File: src/commands/profile/Note.ts (hunk 3 above)
Comment (D0cR3d) [inline src/commands/profile/Note.ts]: Will do.

File: src/commands/profile/Note.ts (hunk 1 above)
Comment (D0cR3d) [inline src/commands/profile/Note.ts]: I'll do this in a future version so we can get this out ASAP.

Comment (katagatame) [inline src/commands/profile/Note.ts]: Sure, just wanna make sure it doesn't get lost.

File: src/commands/profile/Note.ts [hunk 4]
```diff
@@ -10,18 +10,30 @@ export default class Note extends Command {
 	public constructor() {
 		super({
 			name: 'note',
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: after this line add...

`aliases: ['n'],`

File: src/commands/profile/Note.ts [hunk 5]
```diff
@@ -43,78 +55,120 @@ export default class Note extends Command {
 
 		switch (args[0]) {
 			case 'history':
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: before this line add...

`case 'h':`

File: src/commands/profile/Note.ts [hunk 6]
```diff
@@ -43,78 +55,120 @@ export default class Note extends Command {
...
-						console.error(error);
-						message.channel.send(`There was an error while storing the note for ${user.username}.`);
+						});
+					break;
+				} else {
+					message.channel.send(`Sorry, but you do not have the role(s) necessary to access this command. Roles required: \`The Vanguard\` or \`Discord Chat Mods\` or \`Mod Assistant\`.`);
+					return message.channel.stopTyping();
+				}
+
+			case 'add':
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: before this line add...

`case 'a':`

File: src/commands/profile/Note.ts [hunk 7]
```diff
@@ -43,78 +55,120 @@ export default class Note extends Command {
...
-				this.database.commands.notes.get(user.id)
-					.then(results => {
-						if (!results.length) {
-							message.channel.send(`There are no notes for ${user.username}.`);
-							return message.channel.stopTyping();
-						} else {
-							let noteIndex: number = Number(args[2]);
-							if (results.length < noteIndex) {
-								message.channel.send(`Invalid note specified.`);
+			case 'delete':
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: before this line add...

`case 'd':`

File: src/commands/profile/Note.ts [hunk 8]
```diff
@@ -43,78 +55,120 @@ export default class Note extends Command {
...
-								});
-						}
-					})
-					.catch(error => {
-						console.error(error);
-						message.channel.send(`There was an error while deleting a note for ${user.username}.`);
-						return message.channel.stopTyping();
-					});
-				break;
+			case 'reset':
```
Comment (katagatame) [inline src/commands/profile/Note.ts]: before this line add...

`case 'r':`

**Pull request merged**